regolith/helpers/u_milestonehelper.py

```diff
@@ -110,161 +113,177 @@ def construct_global_ctx(self):
                     raise RuntimeError(
-                        "ERROR: This milestone does not have a type set and this is required. "
-                        "Please rerun your command adding '--type' "
-                        f"and typing a type from this list: {MILESTONE_TYPES}")
-                if rc.type:
-                    if rc.type in MILESTONE_TYPES:
-                        doc.update({'type': rc.type})
-                    else:
-                        raise ValueError(
-                            "ERROR: The type you have specified is not recognized. "
-                            "Please rerun your command adding '--type' "
-                            f"and giving a type from this list: {MILESTONE_TYPES}")
-                if rc.finish:
-                    rc.status = "finished"
-                    doc.update({'end_date': now})
-                    notes = doc.get("notes", [])
-                    notes_with_closed_items = [note.replace('()', '(x)', 1) for note in notes]
-                    doc["notes"] = notes_with_closed_items              
-                    if identifier == 'deliverable':
-                        name = 'deliverable'
-                    else:
-                        name = doc['name']
-                    print(
-                        "The milestone {} has been marked as finished in prum {}".format(
-                            name, key))
-                if rc.status:
-                    doc.update({'status': rc.status})
-                if rc.audience:
-                    doc.update({'audience': rc.audience})
-                if rc.due_date:
-                    doc.update({'due_date': rc.due_date})
-                doc['due_date'] = get_due_date(doc)
-                if rc.objective:
-                    doc.update({'objective': rc.objective})
-                if rc.notes:
-                    notes = doc.get("notes", [])
-                    notes.extend(rc.notes)
-                    doc["notes"] = notes
-                if identifier == 'milestones':
-                    if rc.name:
-                        doc.update({'name': rc.name})
-                    new_mil.append(doc)
-                else:
-                    del doc['identifier']
-                    pdoc.update({identifier: doc})
-        new_all = deepcopy(all_milestones)
-        for i, j in zip(index_list, new_all):
-            if j['identifier'] == 'milestones' and i not in idx_parsed:
-                new_mil.append(j)
-        for mile in new_mil:
-            del mile['identifier']
-        new_mil.sort(key=lambda x: x['due_date'], reverse=False)
-        pdoc.update({'milestones': new_mil})
-        rc.client.update_one(rc.database, rc.coll, filterid, pdoc)
-        print("{} has been updated in projecta".format(key))
+                        "Please input a valid projectum id or a valid fragment of a projectum id")
+                print("Projecta not found. Projecta with similar names: ")
```

Comment:
this should probably be "projectum not found, Projecta with similar names....".  I don't think we are ever looking for more than one prum in this workflow.

This also sometimes fails because it is looking in the wrong database, so it is a valid id but the valid id is in say `billingegroup-public` database but `rc.database` is set to `billingegroup-group`.  For now just mention that as a possible reason why this error was raised and later we can think if we can come up with a better fix (could you post that as an issue actually?).   maybe "if you prum id looks correct, check that this id is in the collection in the database that regolith is looking in (i.e., {rc.database})" or sthg like that.

Reply:
I added the additional info to the error statement. Right now the milestone-updater includes an optional argument that allows the user to change `rc.database`. Is that a potential fix? Or were you thinking of a fix that scans multiple databases?

Reply:
please change "projecta" to "projectum" in the first word of the error message.  Yes, this was the fix I was asking for.  Again, to be kinder to the user, it is always better to tell them not so much what is wrong but what to do differently, let's add a statement at the end "...(i.e., {rc.database}).  In this case rerun with --database set to the database where the item is located."   Sthg like that.